examples/step-37/step-37.cc

```diff
@@ -780,51 +780,50 @@ namespace Step37
   {
     Timer time;
     setup_time = 0;
+    {
+      system_matrix.clear();
+      mg_matrices.clear_elements();
 
-    system_matrix.clear();
-    mg_matrices.clear_elements();
-
-    dof_handler.distribute_dofs(fe);
-    dof_handler.distribute_mg_dofs();
-
-    pcout << "Number of degrees of freedom: " << dof_handler.n_dofs()
-          << std::endl;
+      dof_handler.distribute_dofs(fe);
+      dof_handler.distribute_mg_dofs();
 
-    const IndexSet locally_relevant_dofs =
-      DoFTools::extract_locally_relevant_dofs(dof_handler);
+      pcout << "Number of degrees of freedom: " << dof_handler.n_dofs()
+            << std::endl;
 
-    constraints.clear();
-    constraints.reinit(locally_relevant_dofs);
-    DoFTools::make_hanging_node_constraints(dof_handler, constraints);
-    VectorTools::interpolate_boundary_values(
-      mapping, dof_handler, 0, Functions::ZeroFunction<dim>(), constraints);
-    constraints.close();
+      constraints.clear();
+      constraints.reinit(dof_handler.locally_owned_dofs(),
+                         DoFTools::extract_locally_relevant_dofs(dof_handler));
+      DoFTools::make_hanging_node_constraints(dof_handler, constraints);
+      VectorTools::interpolate_boundary_values(
+        mapping, dof_handler, 0, Functions::ZeroFunction<dim>(), constraints);
+      constraints.close();
+    }
```

Comment:
Here, too, I took the liberty of making the code marginally easier to read. The function has three blocks where it starts/stops timers and outputs some timing information for each. To make this easier to see, I've enclosed each block in braces, which changes the indentation but there are no functional changes other than the `AffineConstraints` changes.